### ISSUE TITLE
Now that multiple selection and edition is possible, search should be able to do multi-selection

### ISSUE THREAD

**Issue by cbouc** [notepad-plus-plus/notepad-plus-plus#14420]
### Description of the enhancement
Add a Select button Find dialog to multi-select all occurrences (most often with In selection checked), useful for multi-edition, like selecting separators (commas...), split lines and continue changes from there.

**Comment by alankilborn** [comment]
Probably a duplicate of many other earlier issues...

>search should be able to do multi-selection

True.

Workaround:  Use the _Columns++_ plugin.

**Comment by cbouc** [comment]
Multi selection + edition being a new feature, I searched for find + select and search + select and found no comparable recent issue.

**Comment by alankilborn** [comment]
> Multi selection + edition being a new feature, I searched for find + select and search + select and found no comparable recent issue

It's fine...  
Column-block search/replace has for sure been requested before (column-selection is a form of multi-selection).

**Comment by astewart77** [comment]
> The ask is for search hits to be selected.

Workaround: also in the _Columns++_  plugin is a drop down _Select All_ option from the _Count_ button in the _Search_ dialog.

**Comment by cbouc** [comment]
> It seems I misread what @cbouc was asking for. The ask is for search _hits_ to be selected. I thought, before rereading, that the ask was for searching _in_ multiple selections. Neither thing is currently supported (8.6).
> 
> Supporting multi-select of search hits seems like a good fit as an option in the _Mark_ tab (if this were to be made a native feature of Notepad++).

Yes, the idea is to select all occurrences of searched text (either in whole document or in an existing SINGLE SPAN selection) to be able to apply multi-edition to the occurrences.

**Comment by alankilborn** [comment]
I'll point out that selecting text in such a manner isn't as useful (IMO) as it first might seem.
I mean, you get the text selected -- then what?
Choices are:

* start typing to replace those matches with the same text in each spot (can already do this with a _Replace_ action)
* copy the matches (assuming regex mode because why else would it be useful); can already do this with _Mark_ followed by _Copy Marked Text_
* single press of the delete key to remove the matches (can already do this with a _Replace_ action)
* some follow on careting moves to move to positions _relative_ to the matches

**Comment by molsonkiko** [comment]
> I mean, you get the text selected -- then what?
[JsonTools](https://github.com/molsonkiko/JsonToolsNppPlugin) supports find/replace on each of multiple selections. To the extent that this and other tools can properly handle multiple selections simultaneously, any multi-selection tool gains value.
E.g., select all matches to `\b\w{3}\b` (three-char word) in this file:
```
foo
    bar
  doggone dud  
     baz blood bld
   quz godly
 zut
```
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/46202915/12670cdf-1e71-45f0-95aa-4009e68fd889)
replace `d` with `xy` *in only the selected text* (as shown in the query in the preceding photo), get the following:
```
foo
    bar
  doggone xyuxy  
     baz blood blxy
   quz godly
 zut
```

__I'm aware that it's entirely possible to do many or most things like this using regular expressions alone.__ The Boost regex library that Notepad++ is extremely powerful, and multi-selection followed by find/replace within the multi-selections is probably rarely the best choice. But though I'm a big advocate for regex literacy (and Notepad++ feature literacy), I totally understand why people might prefer the intuitiveness of just typing in a bunch of selections rather than navigating the Notepad++ command menu.
